### PR TITLE
fix: [macOS] Fix PathStreamGeometryContext Arc support

### PR DESCRIPTION
**Pull request by christosk92**
GitHub Issue (If applicable): #

None but has been an issue for a while now. Already notified in discord.

## PR Type

What kind of change does this PR introduce?

- Bugfix

-->

## What is the current behavior?

The radialprogressbar from the community toolkit is simply broken. 
image

## What is the new behavior?

It works now.
image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/Media/PathStreamGeometryContext.cs [hunk 1]
```diff
@@ -133,11 +133,21 @@ public override void ArcTo(Point point, Size size, double rotationAngle, bool is
 			);
 
 #elif __MACOS__
-			bezierPath.AppendPathWithArc(center,
+//Ugly workaround. check if all vars are defined
+			if (!double.IsNaN(radius) && !double.IsNaN(startAngle) && !double.IsNaN(endAngle)) {
+
+				//Convert to degrees in a 0 =< x =< 360 deg range
+				startAngle = (startAngle * (180 / PI) + 360) % 360;
```
Comment (jeromelaban) [inline src/Uno.UI/Media/PathStreamGeometryContext.cs:140]: We already have a helper for that:
```suggestion
				startAngle = MathEx.ToDegreeNormalized(startAngle);
```

File: src/Uno.UI/Media/PathStreamGeometryContext.cs [hunk 2]
```diff
@@ -133,11 +133,21 @@ public override void ArcTo(Point point, Size size, double rotationAngle, bool is
 			);
 
 #elif __MACOS__
-			bezierPath.AppendPathWithArc(center,
+//Ugly workaround. check if all vars are defined
+			if (!double.IsNaN(radius) && !double.IsNaN(startAngle) && !double.IsNaN(endAngle)) {
```
Comment (jeromelaban) [inline src/Uno.UI/Media/PathStreamGeometryContext.cs:137]: What case did you see causes this issue ?

Comment (christosk92) [reply src/Uno.UI/Media/PathStreamGeometryContext.cs:137]: On the initialization process, a NaN would be passed. This caused the arc to not show.
If you resized the window, the arc would be displayed again

**Comment by MartinZikmund** [comment]
@christosk92 one more thing needed before we can merge this - please reword the commit messages adhere to [Conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) (the messages need to be prefixed by "fix:", "chore:" etc.). This can be done very easily from Fork git client for example. Then force-push the changes and it should be good 👍 . If you need help, let me know 🙂

**Comment by christosk92** [comment]
@MartinZikmund I updated the commit names!

**Pull request merged**